### PR TITLE
Add try_put_and_wait documentation

### PR DESCRIPTION
**Pull request by kboyarinov**
### Description 
Add reference page about Flow Graph try_put_and_wait feature

Fixes # - _issue number(s) if exists_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [ ] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [ ] not needed

### Breaks backward compatibility
- [ ] Yes
- [ ] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: doc/main/reference/try_put_and_wait.rst [hunk 1]
```diff
@@ -0,0 +1,324 @@
+.. _try_put_and_wait:
+
+Waiting for single messages in Flow Graph
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:3]:
```suggestion
Waiting for Single Messages in Flow Graph
```

File: doc/main/reference/try_put_and_wait.rst [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+=========================================
+
+.. contents::
+    :local:
+    :depth: 1
+
+Description
+***********
+
+This feature adds a new ``try_put_and_wait`` to the receiving nodes in the Flow Graph.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:13]:
```suggestion
This feature adds a new ``try_put_and_wait`` interface to the receiving nodes in the Flow Graph.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+This function allows to wait until completion of single message, submitted to ``try_put_and_wait`` as an input by the Flow Graph in
+opposite to ``graph.wait_for_all()`` function that waits for completion of all tasks in the graph.
+
+The ``node.try_put_and_wait(msg)`` is performing ``node.try_put(msg)`` on the node and blocks until the work on ``msg`` is completed.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:17]:
```suggestion
``node.try_put_and_wait(msg)`` performs ``node.try_put(msg)`` on the node and blocks it until the work on ``msg`` is completed.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+It means the following two statements:
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:18]:
```suggestion
Therefore, the following conditions are true:
```

File: doc/main/reference/try_put_and_wait.rst [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+
+* Any task spawned by any node in the Flow Graph that corresponds on working with ``msg`` or any other intermediate result
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:20]:
```suggestion
* Any task initiated by any node in the Flow Graph that involves working with msg or any intermediate result derived from ``msg`` is completed.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+computted from ``msg`` is completed.
+* Any intermediate result computted from ``msg`` is retrieved from any buffer in the graph.
+
+.. caution::
+
+    Usage of buffering nodes at the end of the Flow Graph to collect the computation results may prevent ``try_put_and_wait`` calls
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:26]: I'd rephrase to smtn like:

To prevent issues where ``try_put_and_wait`` calls are skipped after the ``wait_for_all call`` completes, do not use buffering nodes at the end of the Flow Graph.

File: doc/main/reference/try_put_and_wait.rst [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+    even after ``wait_for_all`` call would be completed.
+
+.. caution::
+
+    ``multifunction_node`` and ``async_node`` classes are not currently supported by this feature. Having one of these nodes in the
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:31]: The ``multifunction_node`` and ``async_node`` classes are not supported by this feature. Including one of these nodes in the Flow Graph may cause ``try_put_and_wait`` to exit early, even if the computations on the initial input message are still in progress.

File: doc/main/reference/try_put_and_wait.rst [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+
+            template <typename TupleType>
+            class split_node {
+            public:
+                bool try_put_and_wait(const TupleType& input);
+            }; // class split_node
+        } // namespace tbb
+    } // namespace oneapi
+
+Member functions
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:122]:
```suggestion
Member Functions
```

File: doc/main/reference/try_put_and_wait.rst [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+----------------
+
+This section desc
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:125]:
```suggestion

```

File: doc/main/reference/try_put_and_wait.rst [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename Output, typename Policy>
+    bool continue_node<Output, Policy>::try_put_and_wait(const continue_msg& input)
+
+**Effects**: Increments the count of input signals received. If the incremented count is equal to the number
+of known predecessors, performs the ``body`` function object execution.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:135]: Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks associated with each node related to the ``input`` are executed, and no related objects remain in any buffer within the graph.

File: doc/main/reference/try_put_and_wait.rst [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename Input, typename Output, typename Policy>
+    bool function_node<Input, Output, Policy>::try_put_and_wait(const Input& input)
+
+**Effects**: If the concurrency limit allows, executes the user-provided body on the incoming message ``input``.
+Otherwise, depending on the ``Policy`` of the node, either queues the incoming message ``input`` or rejects it.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:148]: same as above

File: doc/main/reference/try_put_and_wait.rst [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true`` if the input was accepted, ``false`` otherwise.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:151]:
```suggestion
**Returns**: ``true`` if the input is accepted, ``false`` otherwise.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 13]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename T>
+    bool overwrite_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Stores ``input`` in the internal single item buffer and broadcasts it to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:160]: same here and later on

File: doc/main/reference/try_put_and_wait.rst [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. caution::
+
+    Since the input element is not retrieved from ``overwrite_node`` once accepted by the successor, the element should be
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:167]:
```suggestion
    Since the input element is not retrieved from ``overwrite_node`` once accepted by the successor, 
```

File: doc/main/reference/try_put_and_wait.rst [hunk 15]
```diff
@@ -0,0 +1,324 @@
...
+    retrieved by explicitly calling the ``clear()`` method or by overwriting by another element to prevent
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:168]:
```suggestion
    retrieve it by explicitly calling the ``clear()`` method or by overwriting with another element to prevent
```

File: doc/main/reference/try_put_and_wait.rst [hunk 16]
```diff
@@ -0,0 +1,324 @@
...
+    ``try_put_and_wait`` from endless waiting.
+
+.. code:: cpp
+
+    template <typename T>
+    bool write_once_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Stores ``input`` in the internal single item buffer if it does not contain a valid value already.
+If a new value is set, the node broadcast it to all successors.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:177]:
```suggestion
If a new value is set, the node broadcasts it to all successors.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 17]
```diff
@@ -0,0 +1,324 @@
...
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. caution::
+
+    Since the input element is not retrieved from ``overwrite_node`` once accepted by the successor, the element should be
+    retrieved by explicitly calling the ``clear()`` method or by overwriting by another element to prevent
+    ``try_put_and_wait`` from endless waiting.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:169]:
```suggestion
    ``try_put_and_wait`` from indefinite waiting.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 18]
```diff
@@ -0,0 +1,324 @@
...
+If a new value is set, the node broadcast it to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true`` for the first time after construction or a call to ``clear()``.
+
+.. caution::
+
+    Since the input element is not retrieved from the ``write_once_node`` once accepted by the successor, the element should
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:186]: same

File: doc/main/reference/try_put_and_wait.rst [hunk 19]
```diff
@@ -0,0 +1,324 @@
...
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. code:: cpp
+
+    template <typename T>
+    bool queue_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the set of items managed by the node, and tries forwarding the least recently added item
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:206]:
```suggestion
**Effects**: Adds ``input`` to the set of items managed by the node and tries forwarding the least recently added item
```

File: doc/main/reference/try_put_and_wait.rst [hunk 20]
```diff
@@ -0,0 +1,324 @@
...
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. code:: cpp
+
+    template <typename T, typename Compare>
+    bool priority_queue_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the ``priority_queue_node`` and tries forwarding to a successor the item with the largest
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:219]: Adds input to the priority_queue_node and attempts to forward the item with the highest priority among all items added to the node but not yet forwarded to the successors.

File: doc/main/reference/try_put_and_wait.rst [hunk 21]
```diff
@@ -0,0 +1,324 @@
...
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. code:: cpp
+
+    template <typename T, typename DecrementType>
+    bool limiter_node<T, DecrementType>::try_put_and_wait(const T& input)
+
+**Effects**: If the broadcast count is below the threshold, ``input`` is broadcasted to all successors.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:244]:
```suggestion
**Effects**: If the broadcast count is below the threshold, broadcasts ``input`` to all successors.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 22]
```diff
@@ -0,0 +1,324 @@
...
+
+    template <typename T>
+    bool broadcast_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Broadcasts ``input`` to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true`` even if the node was unable to successfully forward the message to any of successors.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:261]:
```suggestion
**Returns**: ``true`` even if the node cannot successfully forward the message to any of its successors.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 23]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename TupleType>
+    bool split_node<TupleType>::try_put_and_wait(const TupleType& input);
+
+**Effects**: Broadcasts each element in the incoming tuple to the nodes connected to the ``split_node`` output ports.
+The element at index ``i`` of ``input`` will be broadcasted through the output port number ``i``.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:269]:
```suggestion
The element at index ``i`` of ``input`` is broadcasted through the output port number ``i``.
```

File: doc/main/reference/try_put_and_wait.rst (hunk 1 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 2 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 3 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Partially fixed, I would prefer to use `waiting until` instead of `blocks` since it is not clear what "blocking the function" means and I don't want to use terminology of blocking the thread here.

File: doc/main/reference/try_put_and_wait.rst (hunk 4 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 5 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Partially fixed, using `derived from` can be ambiguous with the C++ inheritance terminology. Can we keep `computed` here?

File: doc/main/reference/try_put_and_wait.rst (hunk 6 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Rewritten.

File: doc/main/reference/try_put_and_wait.rst (hunk 7 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed.

File: doc/main/reference/try_put_and_wait.rst (hunk 9 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Removed

File: doc/main/reference/try_put_and_wait.rst (hunk 12 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 13 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 14 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 16 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 10 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 17 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 19 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 18 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 20 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 21 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 23 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 22 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 5 above)
Comment (aepanchi) [reply doc/main/reference/try_put_and_wait.rst]: yes, computed is fine

File: doc/main/reference/try_put_and_wait.rst [hunk 24]
```diff
@@ -0,0 +1,322 @@
...
+
+.. contents::
+    :local:
+    :depth: 1
+
+Description
+***********
+
+This feature adds a new ``try_put_and_wait`` interface to the receiving nodes in the Flow Graph.
+This function allows to wait until completion of single message, submitted to ``try_put_and_wait`` as an input by the Flow Graph in
```
Comment (vossmjp) [inline doc/main/reference/try_put_and_wait.rst:14]: I'd suggest, "This function puts a message as an input into a Flow Graph and then waits until all work related to that input is complete."

Comment (vossmjp) [reply doc/main/reference/try_put_and_wait.rst:14]: I'd add a second sentence, "try_put_and_wait may reduce latency compared to calling graph::wait_for_all since graph::wait_for_all waits for all work, including work unrelated to the input message, to complete."

File: doc/main/reference/try_put_and_wait.rst [hunk 25]
```diff
@@ -0,0 +1,322 @@
...
+opposite to ``graph.wait_for_all()`` function that waits for completion of all tasks in the graph.
+
+``node.try_put_and_wait(msg)`` performs ``node.try_put(msg)`` on the node and waits until the work on ``msg`` is completed.
+Therefore, the following conditions are true:
+
+* Any task initiated by any node in the Flow Graph that involves on working with ``msg`` or any other intermediate result
```
Comment (vossmjp) [inline doc/main/reference/try_put_and_wait.rst:20]:
```suggestion
* Any task initiated by any node in the Flow Graph that involves working with ``msg`` or any other intermediate result
```

File: doc/main/reference/try_put_and_wait.rst [hunk 26]
```diff
@@ -0,0 +1,322 @@
...
+computed from ``msg`` is completed.
+* Any intermediate result computed from ``msg`` is retrieved from any buffer in the graph.
```
Comment (vossmjp) [inline doc/main/reference/try_put_and_wait.rst:22]:
```suggestion
* No intermediate results computed from ``msg`` remain in any buffers in the graph.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 27]
```diff
@@ -0,0 +1,322 @@
...
+
+.. caution::
+
+    To prevent ``try_put_and_wait`` calls from infinite blocking, avoid using buffering nodes at the end of the Flow Graph since the final result
+    would not be retrieved by the Flow Graph.
```
Comment (vossmjp) [inline doc/main/reference/try_put_and_wait.rst:27]:
```suggestion
    will not be automatically consumed by the Flow Graph.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 28]
```diff
@@ -0,0 +1,322 @@
...
+``node.try_put_and_wait(msg)`` performs ``node.try_put(msg)`` on the node and waits until the work on ``msg`` is completed.
+Therefore, the following conditions are true:
+
+* Any task initiated by any node in the Flow Graph that involves on working with ``msg`` or any other intermediate result
+computed from ``msg`` is completed.
+* Any intermediate result computed from ``msg`` is retrieved from any buffer in the graph.
+
+.. caution::
+
+    To prevent ``try_put_and_wait`` calls from infinite blocking, avoid using buffering nodes at the end of the Flow Graph since the final result
```
Comment (vossmjp) [inline doc/main/reference/try_put_and_wait.rst:26]:
```suggestion
    To prevent ``try_put_and_wait`` calls from infinite waiting, avoid using buffering nodes at the end of the Flow Graph since the final result
```

File: doc/main/reference/try_put_and_wait.rst [hunk 29]
```diff
@@ -0,0 +1,322 @@
...
+
+        // Submit work into the graph
+        parallel_for(0, 100, [](int input) {
+            start_node.try_put_and_wait(input);
+
+            // Post processing the result of input
+        });
+    }
+
+Each iteration of ``parallel_for`` submits a part of work into the Flow Graph. After returning from ``try_put_and_wait(input)``, it is
```
Comment (vossmjp) [inline doc/main/reference/try_put_and_wait.rst:320]:
```suggestion
Each iteration of ``parallel_for`` submits an input into the Flow Graph. After returning from ``try_put_and_wait(input)``, it is
```

**Review by vossmjp — COMMENTED** [review]
I think the wording for "indefinite waiting" should be used consistently. So maybe double check that don't match that language are intentional, for example check for "blocking" and "infinite waiting".

File: doc/main/reference/try_put_and_wait.rst (hunk 24 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 25 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 27 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 29 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 26 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst [hunk 30]
```diff
@@ -0,0 +1,324 @@
...
+    :depth: 1
+
+Description
+***********
+
+This feature adds a new ``try_put_and_wait`` interface to the receiving nodes in the Flow Graph.
+This function puts a message as an input into a Flow Graph and then waits until all work related to
+that message is complete.
+``try_put_and_wait`` may reduce latency compared to calling ``graph::wait_for_all`` since
+``graph::wait_for_all`` waits for all work, including unrelated to the input message, to complete.
```
Comment (vossmjp) [inline doc/main/reference/try_put_and_wait.rst:17]:
```suggestion
``graph::wait_for_all`` waits for all work, including work that is unrelated to the input message, to complete.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 31]
```diff
@@ -0,0 +1,324 @@
...
+
+``node.try_put_and_wait(msg)`` performs ``node.try_put(msg)`` on the node and waits until the work on ``msg`` is completed.
+Therefore, the following conditions are true:
+
+* Any task initiated by any node in the Flow Graph that involves working with ``msg`` or any other intermediate result
+computed from ``msg`` is completed.
+* No intermediate results computed from ``msg`` remain in any buffers in the graph.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:24]: Didn't render correctly:
image

File: doc/main/reference/try_put_and_wait.rst [hunk 32]
```diff
@@ -0,0 +1,324 @@
...
+
+.. contents::
+    :local:
+    :depth: 1
+
+Description
+***********
+
+This feature adds a new ``try_put_and_wait`` interface to the receiving nodes in the Flow Graph.
+This function puts a message as an input into a Flow Graph and then waits until all work related to
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:14]:
```suggestion
This function puts a message as an input into a Flow Graph and waits until all work related to
```

File: doc/main/reference/try_put_and_wait.rst [hunk 33]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename Output, typename Policy>
+    bool continue_node<Output, Policy>::try_put_and_wait(const continue_msg& input)
+
+**Effects**: Increments the count of input signals received. If the incremented count is equal to the number
+of known predecessors, performs the ``body`` function object execution.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:135]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 34]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename Input, typename Output, typename Policy>
+    bool function_node<Input, Output, Policy>::try_put_and_wait(const Input& input)
+
+**Effects**: If the concurrency limit allows, executes the user-provided body on the incoming message ``input``.
+Otherwise, depending on the ``Policy`` of the node, either queues the incoming message ``input`` or rejects it.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:148]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 35]
```diff
@@ -0,0 +1,324 @@
...
+related to ``input`` are executed, and no related objects remain in any buffer within the graph.
+
+**Returns**: ``true`` if the input is accepted, ``false`` otherwise.
+
+.. code:: cpp
+
+    template <typename T>
+    bool overwrite_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Stores ``input`` in the internal single item buffer and broadcasts it to all successors.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:158]:
```suggestion
**Effects**: Stores ``input`` in the internal single-item buffer and broadcasts it to all successors.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 36]
```diff
@@ -0,0 +1,324 @@
...
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:160]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 37]
```diff
@@ -0,0 +1,324 @@
...
+    Since the input element is not retrieved from ``overwrite_node`` once accepted by the successor,
+    retrieve it by explicitly calling the ``clear()`` method or by overwriting with another element to prevent
+    ``try_put_and_wait`` from indefinite waiting.
+
+.. code:: cpp
+
+    template <typename T>
+    bool write_once_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Stores ``input`` in the internal single item buffer if it does not contain a valid value already.
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:176]:
```suggestion
**Effects**: Stores ``input`` in the internal single-item buffer if it does not contain a valid value already.
```

File: doc/main/reference/try_put_and_wait.rst [hunk 38]
```diff
@@ -0,0 +1,324 @@
...
+If a new value is set, the node broadcasts it to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:179]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 39]
```diff
@@ -0,0 +1,324 @@
...
+    retrieve it by explicitly calling the ``clear()`` method to prevent ``try_put_and_wait`` from indefinite waiting.
+
+.. code:: cpp
+
+    template <typename T>
+    bool buffer_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the set of items managed by the node and tries forwarding it to a successor.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:196]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 40]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename T>
+    bool queue_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the set of items managed by the node and tries forwarding the least recently added item
+to a successor.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:209]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 41]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename T, typename Compare>
+    bool priority_queue_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the ``priority_queue_node`` and attempts to forward the item with the highest
+priority among all items added to the node but not yet forwarded to the successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:222]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 42]
```diff
@@ -0,0 +1,324 @@
...
+**Returns**: ``true``.
+
+.. code:: cpp
+
+    template <typename T>
+    bool sequencer_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the ``sequencer_node`` and tries forwarding the next item in sequence to a successor.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:234]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 43]
```diff
@@ -0,0 +1,324 @@
...
+**Returns**: ``true``.
+
+.. code:: cpp
+
+    template <typename T, typename DecrementType>
+    bool limiter_node<T, DecrementType>::try_put_and_wait(const T& input)
+
+**Effects**: If the broadcast count is below the threshold, broadcasts ``input`` to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:246]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 44]
```diff
@@ -0,0 +1,324 @@
...
+**Returns**: ``true`` if ``input`` is broadcasted; ``false`` otherwise.
+
+.. code:: cpp
+
+    template <typename T>
+    bool broadcast_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Broadcasts ``input`` to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:258]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst [hunk 45]
```diff
@@ -0,0 +1,324 @@
...
+
+.. code:: cpp
+
+    template <typename TupleType>
+    bool split_node<TupleType>::try_put_and_wait(const TupleType& input);
+
+**Effects**: Broadcasts each element in the incoming tuple to the nodes connected to the ``split_node`` output ports.
+The element at index ``i`` of ``input`` is broadcasted through the output port number ``i``.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks created by each node and
```
Comment (aepanchi) [inline doc/main/reference/try_put_and_wait.rst:271]:
```suggestion
Waits for the completion of the ``input`` in the Flow Graph, meaning all tasks created by each node and
```

File: doc/main/reference/try_put_and_wait.rst (hunk 30 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 31 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst:24]: Fixed

File: doc/main/reference/try_put_and_wait.rst (hunk 32 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 33 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 34 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 36 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 37 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 39 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 40 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 41 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 42 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 43 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 44 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

File: doc/main/reference/try_put_and_wait.rst (hunk 45 above)
Comment (kboyarinov) [reply doc/main/reference/try_put_and_wait.rst]: Done

**Review by aepanchi — APPROVED** [review]
